### PR TITLE
Migrate Kubebuilder from v2 to v3

### PR DESCRIPTION
**Pull request by danielgospodinow**
**Description**

Update Kubebuilder version from v2 to v3 using [this](https://book.kubebuilder.io/migration/migration_guide_v2tov3.html) guideline.

**Pull Request status**

- [x] Implementation
- [x] Unit tests
- [x] Integration tests
- [x] `chart/compass/values.yaml` is updated 
- [x] Mocks are regenerated, using the automated script

### CODE REVIEW

File: chart/compass/charts/operations-controller/templates/rbac.yaml [hunk 1]
```diff
@@ -30,6 +30,18 @@ rules:
...
+  resources:
+  - leases
+  verbs:
+  - get
+  - list
+  - watch
+  - create
+  - update
+  - patch
+  - delete
```
Comment (PetarTodorovv) [inline chart/compass/charts/operations-controller/templates/rbac.yaml:44]: What is the purpose of these new leases resources and is this configuration auto-generated or manually added?

**Review by PetarTodorovv — COMMENTED** [review]
The readme file is outdated, there aren't steps like install/uninstall anymore. Most probably it's not adapted with the previous PRs

**Comment by PetarTodorovv** [comment]
Do we need scripts/bin/kustomize binary file and what's its purpose?

**Comment by PetarTodorovv** [comment]
Just wondering what approach did you use to upgrade kubebuilder version? Manually updating files or generate new v3 project and replacing files?

File: chart/compass/charts/operations-controller/templates/rbac.yaml (hunk 1 above)
Comment (danielgospodinow) [reply chart/compass/charts/operations-controller/templates/rbac.yaml:44]: Those `leases` are a Kubernetes concept for concurrency management. The standard generated RBAC settings from the new v3 Kubebuilder were containing this additional `leases` resource. Without it, the controller doesn't function correctly.

**Comment by danielgospodinow** [comment]
> Do we need scripts/bin/kustomize binary file and what's its purpose?

You mean the `./bin/kustomize` file I suppose?

Its purpose is the same as the `./bin/controller-gen` binary that we currently have. The idea is to cache locally the binary so to speed up certain Makefile targets. This optimization was already available for `controller-gen` in Kubebuilder v2, now, in v3, it's also available for `kustomize`.

**Comment by danielgospodinow** [comment]
> Just wondering what approach did you use to upgrade kubebuilder version? Manually updating files or generate new v3 project and replacing files?

The chosen migration approach was the recommended one from the migration guideline.

> "The recommended way to migrate a v2 project is to create a new v3 project and copy over the API and the reconciliation code.", [Kubebuilder Migration Guide](https://book.kubebuilder.io/migration/migration_guide_v2tov3.html)

**Comment by danielgospodinow** [comment]
> The readme file is outdated, there aren't steps like install/uninstall anymore. Most probably it's not adapted with the previous PRs

Yes, absolutely, I'll refresh the README.

**Comment by PetarTodorovv** [comment]
Looks good to me ✅

**Review by ekaterina-mitova — DISMISSED** [review]

**Review by PetarTodorovv — APPROVED** [review]

**Review by ekaterina-mitova — APPROVED** [review]

**Pull request merged**